datafusion/common/src/encryption.rs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Support optional features for encryption in Parquet files.
+//! This module provides types and functions related to encryption in Parquet files.
+
+#[cfg(feature = "parquet_encryption")]
```

Comment:
Rather than having so may `cfg` lines in this file, you could have the module implemented in two separate files, one for when encryption is enabled and one for when it's disabled.

Eg. see https://github.com/apache/arrow-rs/blob/ff3a2f2c59f0355f8afedb3e9258e1d6307f21ae/parquet/src/column/mod.rs#L121-L125

Reply:
Thanks. That's an interesting idea, but I will stick with the current approach for now.
1. The file and functions are relatively simple, at under 100 lines long.
2. Having two separate files, which may or may not be used, is somewhat confusing.
3. I don't like the idea of having to maintain the same API in two separate places/files. Here, the two versions are displayed side by side, making it straightforward to update them simultaneously.